acceptance/tests/parser_functions/hiera_in_templates.rb

```diff
@@ -257,23 +266,43 @@ class #{@module_name}::mod_fqdn {
 end
 
 def find_osfamilies
-  osfamilies = []
+  family_hash = {}
   agents.each do |agent|
     res = on(agent, facter("osfamily"))
     osf = res.stdout.chomp
-    osfamilies += [osf]
+    family_hash[osf] = 1
+  end
+  osfamilies = family_hash.keys
+end
+
+def find_tmp_dirs
+  tmp_dirs = ""
+  host_to_result_dir = {}
+  agents.each do |agent|
+    h = on(agent, facter("hostname")).stdout.chomp
+    t = agent.tmpdir("#{@module_name}_results")
+    tmp_dirs += "  #{h}: '#{t}'\n"
+    host_to_result_dir[h] = t
   end
-    osfamilies
+  result = {
+    'tmp_dirs' => tmp_dirs,
+    'host_to_result_dir' => host_to_result_dir
+  }
+  result
```

Comment:
maybe this is where josh was saying to return two objects?
i'm okay with it either way.  this test is pretty complex as-is.